File: python/2017/2/solution2.py
from itertools import combinations

with open('../../../data/2017/2/data.txt', 'rb') as f:
    rows = [map(int, row.strip().split('\t')) for row in f.readlines()]

total = 0
for row in rows:
    for pair in combinations(row,2):
        if max(pair) % min(pair) == 0:
            total += max(pair) / min(pair)
        else:
            pass

print "Solution is %d" % total

File: julia/2017/2/solution2.jl
using Combinatorics

f = open("../../../data/2017/2/data.txt")
lines = readlines(f)
close(f)

total = 0
for line in lines
    splitline = map(x->parse(Int,x), split(line, "\t"))
    for pair in combinations(splitline, 2)
        if maximum(pair) % minimum(pair) == 0
            total += maximum(pair) / minimum(pair)
        end
    end
end

println("Solution is $total")
